File: intelligent_financial_assistant_backend/recipient-service/src/main/java/com/khaoula/recipientservice/service/RecipientService.java
package com.khaoula.recipientservice.service;

import com.khaoula.recipientservice.model.Recipient;

import java.util.List;
import java.util.Optional;

public interface RecipientService {
    List<Recipient> getRecipientList(Long userId);
    Recipient addRecipient(Recipient recipient, Long userId);
    Recipient updateRecipient(Long id, Recipient recipient, Long userId);
    void deleteRecipient(Long id, Long userId);
    Optional<Recipient> getByIban(String iban);
}

File: intelligent_financial_assistant_frontend/lib/features/recipient/domains/services/recipient_service.dart
import '../../../../data/response/api_response.dart';
import '../repositories/recipient_repository_interface.dart';
import 'recipient_service_interface.dart';

class RecipientService implements RecipientServiceInterface {
  final RecipientRepositoryInterface recipientRepository;

  RecipientService({required this.recipientRepository});

  @override
  Future<ApiResponse> getRecipientList() async {
    return await recipientRepository.getRecipientList();
  }

  @override
  Future<ApiResponse> addRecipient(String fullName, String iban) async {
    return await recipientRepository.addRecipient(fullName, iban);
  }

  @override
  Future<ApiResponse> updateRecipient(int id, String fullName, String iban) async {
    return await recipientRepository.updateRecipient(id, fullName, iban);
  }

  @override
  Future<ApiResponse> deleteRecipient(int id) async {
    return await recipientRepository.deleteRecipient(id);
  }
}